caluma/caluma_workflow/models.py

```diff
@@ -253,6 +260,27 @@ class WorkItem(UUIDModel):
         null=True,
     )
 
+    def get_redoable(self):
+        jexl = None
+        if self.previous_work_item:
```

Comment:
Why are we looking at the `previous_work_item`'s redoable field instead of "our own" here?

Also: if you'd invert the `if` condition, you could simplify the code by using an early return

Reply:
> Why are we looking at the previous_work_item's redoable field instead of "our own" here?

Because there we have the `redoable` field. If we would set it on our own task_flow, we wouldn't be able to redo a workitem if we reached one without a task_flow (the last one in a branch).

> Also: if you'd invert the if condition, you could simplify the code by using an early return

Done

Reply:
> > Why are we looking at the previous_work_item's redoable field instead of "our own" here?
> 
> Because there we have the `redoable` field. If we would set it on our own task_flow, we wouldn't be able to redo a workitem if we reached one without a task_flow (the last one in a branch).

Aah, that makes sense. However, I'm really not too happy about that. I think this really should be directly on the affected work item ('s task flow) instead. The mental indirection required sounds rather tricky IMO. Consider that a TaskFlow tells us "where to go from here" (`flow.next`), I would strongly expect the `redo` to tellus "where to go back from here", not "where to go back from the *next* item".

This unfortunately sounds like a bit more work, but I think we should store the "redo" data local to where it's relevant, not "on the previous task's task flow".

One solution I think would be acceptable for this is to make the `flow` foreign key nullable, so we can have a task flow that doesn't lead anywhere but may allow redoing. In the long term, as discussed, the `Flow` may even be dropped alltogether

Reply:
Done